sony_build_symlinks.mk

```diff
@@ -0,0 +1,16 @@
+# Creates symlinks relative to the PRODUCT_OUT folder.
+# This is different from BOARD_ROOT_EXTRA_SYMLINKS in that it allows symlinks on other partitions.
+#
+# Links can be specified in TARGET:DIRECTORY or TARGET:LINK_NAME notation. This creates a link named LINK_NAME or DIRECTORY/$(basename TARGET) pointing to TARGET.
+
+ifeq ($(SONY_SYMLINKS),)
+    $(warning "No symlinks set!")
+else
+LOCAL_POST_INSTALL_CMD := $(foreach s,$(SONY_SYMLINKS), \
+    $(eval p := $(subst :,$(space),$(s))) \
```

Comment:
Why not use a space here in the first place?

Reply:
well, using `$(space)` is easier to read.

the code here is taken from `system/core/rootdir/Android.mk`, and they use `$(space)`.

Reply:
re-reading your comment, I guess you meant why not use inputs of the form `t1 n1  t2 n2 t3 n3 ...` instead of `t1:n1  t2:n2 t3:n3 ...`

Reply:
Yup, unless whitespace has some special meaning here, it seems more sensible to just do that directly.
In the context of `BOARD_ROOT_EXTRA_SYMLINKS` Google decided to use the `SOURCE:TARGET` syntax and then convert that to `SOURCE TARGET` here, but since we are contructing the file list automatically, we can just skip that step.

If however we want to add `SOURCE:TARGET` pairings manually later, it would make sense to keep Google's convention.

Reply:
You are both missing the fact that the initial foreach loops over SONY_SYMLINKS with multiple TARGET:LINK_NAME pairs separated by spaces. If the `TARGET` and `LINK_NAME` are also separated by spaces, you need to add a lot of extra trickery to combine every second element with the first. This is why `:` is used to group the two together.

Reply:
I didn't miss that fact :-)

In any case, my preference is to leave it as it is.  The colon-separator is a familiar pattern in aosp makefiles.